### PR TITLE
Update docs URL

### LINKED ISSUE #3: Update docs url
Doc url has moved to tskit.dev

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Summary: Infer tree sequences
 
 Development: https://github.com/tskit-dev/tsinfer
 
-Documentation: https://tsinfer.readthedocs.io/en/stable
```
Comment (jeromekelleher) [inline README.md:14]: I think this gets updated automatically when the recipe is rerendered.

**Comment by jeromekelleher** [comment]
@conda-forge-admin please rerender

File: README.md (hunk 1 above)
Comment (jeromekelleher) [reply README.md:14]: Just rerendered and yes. Doesn't matter, just fyi

**Review by jeromekelleher — APPROVED** [review]
LGTM. Should we push out the new packages (increment the build number) or not bother until next release?

**Comment by jeromekelleher** [comment]
On reflection we should update the packages so we see if there's any problems with the rerendering before the next release.

Can you bump the build number please @benjeffery?

**Comment by benjeffery** [comment]
> Can you bump the build number please @benjeffery?

Done

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**